File: benchmarks/benchmarks/02_numerical_computation/comparisons/matrix_multiply.cpp
#include <iostream>

static long long matrix_checksum(long long size) {
    long long total = 0;
    for (long long row = 0; row < size; ++row) {
        for (long long col = 0; col < size; ++col) {
            long long cell = 0;
            for (long long inner = 0; inner < size; ++inner) {
                cell += ((row * inner) + 3) * ((inner * col) + 5);
            }
            total += cell % 104729;
        }
    }
    return total;
}

int main() {
    std::cout << matrix_checksum(64) << '\n';
    return 0;
}

File: benchmarks/benchmarks/02_numerical_computation/comparisons/matrix_multiply.rs
fn matrix_checksum(size: i64) -> i64 {
    let mut total = 0;
    for row in 0..size {
        for col in 0..size {
            let mut cell = 0;
            for inner in 0..size {
                cell += ((row * inner) + 3) * ((inner * col) + 5);
            }
            total += cell % 104_729;
        }
    }
    total
}

fn main() {
    println!("{}", matrix_checksum(64));
}

